test/functional/uplink.auth.spec.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const uplinkStorage = require('../../src/lib/storage/up-storage');
+const assert = require('assert');
+
+function createUplink(config) {
+  const defaultConfig = {
+    url: 'https://registry.npmjs.org/'
+  };
+  let mergeConfig = Object.assign({}, defaultConfig, config);
+  return new uplinkStorage(mergeConfig, {});
+}
+
+function setHeaders(config, headers) {
+  config = config || {};
+  headers = headers || {};
+  const uplink = createUplink(config);
+  return uplink._setHeaders({
+    headers
+  });
+}
+
+module.exports = function () {
+
+  describe('uplink auth test', function () {
+
+    it('if set headers empty', function () {
+      const headers = setHeaders();
+
+      assert.equal(Object.keys(headers).length, 3);
+    });
+
+    it('invalid auth', function () {
+      const fnError = function () {
+        setHeaders({
+          auth: ''
+        });
+      };
+
+      assert.throws(fnError, 'Auth invalid');
+    });
+
+    it('if set headers authorization', function () {
+      const headers = setHeaders({}, {
+        'authorization': 'basic Zm9vX2Jhcg=='
+      });
+      assert.equal(Object.keys(headers).length, 4);
```

Comment:
Just to be consistent, you left break lines between assets and assignment in some test which is great. Could you do in the others as well?